src/components/veNft/veNftCarousel.tsx

```diff
@@ -1,5 +1,114 @@
-const veNftCarousel = () => {
-  return <div>veNftCarousel</div>
+import { Row, Col, Image, FormInstance } from 'antd'
+import { VeNftVariant } from 'models/v2/veNft'
+import { getVeNftBaseImage } from 'utils/v2/veNft'
+
+type StakingNFTCarouselProps = {
+  tokensStaked: string
+  variants: VeNftVariant[]
+  baseImagesHash: string
+  form: FormInstance
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  tokenMetadata: any
 }
 
-export default veNftCarousel
+export default function StakingNFTCarousel({
+  form,
+  tokensStaked,
+  variants,
+  baseImagesHash,
+  tokenMetadata,
+}: StakingNFTCarouselProps) {
+  const activeIdx = variants
+    ? Math.max(
+        variants.findIndex(variant => {
+          const curTokens = parseInt(tokensStaked)
+          return (
+            variant.tokensStakedMin <= curTokens &&
+            (variant.tokensStakedMax
+              ? variant.tokensStakedMax > curTokens
+              : true)
+          )
+        }),
+        0,
+      )
+    : 0
+
+  const prevVariant = activeIdx - 1 < 0 ? undefined : variants[activeIdx - 1]
+  const prevVariantImage = prevVariant
+    ? getVeNftBaseImage(baseImagesHash, prevVariant)
+    : undefined
+  const currentVariant = variants[activeIdx]
+  const currentImage = tokenMetadata
+    ? tokenMetadata.thumbnailUri
+    : getVeNftBaseImage(baseImagesHash, currentVariant)
+  const nextVariant =
+    activeIdx + 1 >= variants.length ? undefined : variants[activeIdx + 1]
+  const nextVariantImage = nextVariant
+    ? getVeNftBaseImage(baseImagesHash, nextVariant)
+    : undefined
+
+  const nonActiveStyle = { opacity: 0.3 }
+
+  const handleImageClick = (variant: VeNftVariant) => {
+    form.setFieldsValue({ tokensStaked: variant.tokensStakedMin })
+  }
+
+  const dims = 150
+
+  const variantRangeInfo = (variant: VeNftVariant, opacity?: number) => {
+    return (
+      <p style={{ textAlign: 'center', opacity: opacity }}>
+        {variant.tokensStakedMin}-
+        {variant.tokensStakedMax ? variant.tokensStakedMax : '+'}
+      </p>
+    )
+  }
+
+  return (
+    <Row>
+      <Col span={8}>
+        {prevVariant && (
+          <>
+            <div onClick={() => handleImageClick(prevVariant)}>
```

Comment:
**question** do we need this div? could we potentially add the onClick handler to `Image` instead?